src/scala/Eq.scala

```diff
@@ -0,0 +1,23 @@
+package scala
+
+import annotation.implicitNotFound
+
+/** A marker trait indicating that values of kind `T` can be compared to values of type `U`. */
+@implicitNotFound("Values of types ${L} and ${R} cannot be compared with == or !=")
+trait Eq[-L, -R]
```

Comment:
I think this trait should be `sealed` since the only instance should be the object `Eq`